### PR TITLE
CI job with minimal dependencies

### PR DESCRIPTION
**Pull request by volkm**
Added CI job with minimal dependencies based on the Docker image `storm-basesystem:minimal_dependencies`:
- without Z3 (as suggested in https://github.com/moves-rwth/storm/issues/282)
- without Eigen
- without xerces
- without CLN and GiNaC
- without GLPK
- without hwloc

The CI job already detected multiple issues:
- [x] shipped Carl did not compile without CLN and GiNaC
- [x] minimal build of Carl needs changes of https://github.com/moves-rwth/carl-storm/pull/40. In particular, the patch step of Eigen in the shipped Carl requires a valid git committer.
- [x] a larger number of tests are failing due to missing SMT solver
- [x] Glpk needed reconfiguration after removing examples folder

### CODE REVIEW

**Comment by tquatmann** [comment]
current state LGTM!

File: resources/3rdparty/glpk-5.0/Makefile.am [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 ACLOCAL_AMFLAGS=-I m4
 
-SUBDIRS = src examples
```
Comment (tquatmann) [inline resources/3rdparty/glpk-5.0/Makefile.am:5]: We probably should add glpk to the `update_resources.md` and document what you have done to get rid of the examples folder.

**Review by tquatmann — APPROVED** [review]
LGTM! Just one minor comment.

File: resources/3rdparty/glpk-5.0/Makefile.am (hunk 1 above)
Comment (volkm) [reply resources/3rdparty/glpk-5.0/Makefile.am:5]: Good point, thanks for the comment.
I added the documentation.

**Pull request merged**